### ISSUE TITLE
TypeError when using methods of imported es6 classes

### ISSUE THREAD

**Issue by 0xR** [babel/babel-jest#22]
When I have a file `undertest.js`:
``` javascript
import SimpleClass from './SimpleClass';

(new SimpleClass()).hello();
```
And `SimpleClass.js`:
```javascript
export default class SimpleClass {
  hello(){
    console.log('hello world')
  }
}
```
A test file only containing this line:
``` javascript
require('../undertest');
```
I get this error:
```
TypeError: /some/path/src/lib/__tests__/test.js: /some/path/src/lib/undertest.js: Object function SimpleClass() {
    _classCallCheck(this, SimpleClass);
  } has no method 'hello'
1 test failed, 0 tests passed (1 total)
Run time: 0.867s
```
I am running jest from gulp with babel-jest like this:
```javascript
gulp.task('jest', function(done) {
  var rootDir = './src';
  jest.runCLI({config: {
    'rootDir': rootDir,
    'scriptPreprocessor': '../node_modules/babel-jest',
    'testFileExtensions': ['es6', 'js'],
    'moduleFileExtensions': ['js', 'json', 'es6']
  }}, rootDir, function(success) {
    done(success ? null : 'jest failed');
    process.on('exit', function() {
      process.exit(success ? 0 : 1);
    });
  });
});
```

I am running node `v0.10.38` and I removed `node_modules` and reinstalled since switching from node latest (normally I just do a `npm rebuild`). Babel-jest ^5.0.0, babel ^5.0.8.

See https://github.com/0xR/este/tree/babel-jest-error for an example.

**Comment by sebmck** [comment]
Probably because the class methods are non-enumerable and Jest hasn’t been updated.

**Comment by 0xR** [comment]
So I have to wait for Jest to update? Since I am on the latest version of Jest.

**Comment by sebmck** [comment]
Likely related to facebook/jest#332

**Comment by timarney** [comment]
Not sure if this is the issue but I just ran into the same thing.  I added the "unmockedModulePathPatterns" line to my package.json and that seems to have fixed things.  

```javascript

"jest": {
    "scriptPreprocessor": "<rootDir>/node_modules/babel-jest",
    "unmockedModulePathPatterns": ["<rootDir>/node_modules/react"],
    "testFileExtensions": [
      "es6",
      "js"
    ],
    "moduleFileExtensions": [
      "js",
      "json",
      "es6"
    ]
  },

```

**Comment by ericwooley** [comment]
Any update for this? facebook/jest#332 is closed, and I am getting errors on a lot of es6 things

~~~javascript
class SpecialRecord extends Immutable.Record({
  test: Immutable.Seq(),
  test2: Immutable.Seq()
}) {
  constructor(obj = {}) {
    //... modifications to object 
    super(obj)
  }
}
~~~

Gives me this error: Super expression must either be null or a function, not undefined

I tried it with jest.autoMockOff() in the beginning of my test also, with no change :(

**Comment by arush** [comment]
I can confirm this is a babel-jest issue, when i change my require() to es6 style import, jest incorrectly mocks the import even though i've set it to .dontMock(). Changing back to require() works fine

**Comment by laposheureux** [comment]
Additional confirm, since we're not using require, we've needed to place more things in the jest config in package.json in the `unmockedModulePathPatterns` array.

**Comment by laposheureux** [comment]
And this is the workaround: https://github.com/babel/babel-jest/issues/16#issuecomment-155611005 because imports are hoisted so they execute before the dontMock function calls.

**Comment by jamedranoa** [comment]
I think this might be related https://github.com/facebook/jest/issues/693

**Comment by cpojer** [comment]
This has been fixed. The dontMock and import hoisting issues will also be resolved in 0.9.0, see https://github.com/facebook/jest/pull/741
